### DISCUSSION TITLE
encrypt method return type error

### DISCUSSION THREAD

**Discussion by clarencedo** [openpgpjs/openpgpjs#1759] [Q&A] [answered]
OpenPGP.js version: 5.10.2
Affected platform (Browser or Node.js version): Both, Typescript

I use the encrypt method, and the return type is alway Promise<openpgp.Webstream<T>> instead of Promise<T>.

`    
const encrypted = await openpgp.encrypt({
            message: pgpMessage,
            format: "armored",
            encryptionKeys: pgpPublicKey,
            signingKeys: pgpPrivateKey,
            config: {
                preferredCompressionAlgorithm: openpgp.enums.compression.zip,
                preferredSymmetricAlgorithm: openpgp.enums.symmetric.aes256,
                preferredHashAlgorithm: openpgp.enums.hash.sha256,
            },
        });
`

if the format = 'armored' , return type is Promise<openpgp.WebStream<string>>
 instead of 'string'
if the format = 'binary', return type is Promise<openpgp.WebStream<Uint8Array>> 
instead of 'Uint8Array'
I don't know why always return openpgp.WebStream<T>

**Comment by larabr** [comment, marked as answer]
Hi, how is `pgpMessage` declared?
Also, did you install the correct TS deps as explained [here](https://github.com/openpgpjs/openpgpjs?tab=readme-ov-file#typescript)?

**Reply by clarencedo** [reply]
truly appreciate, this was caused by missing TS deps.
